### PR TITLE
Server-side implementation of batched logging endpoints

### PR DESCRIPTION
**Pull request by smurching**
This PR is a follow-up to #955 adding support for logging batches to a remote server, as well as server-side handlers for batched logging, and associated unit tests.

### CODE REVIEW

File: mlflow/server/handlers.py [hunk 1]
```diff
@@ -330,6 +330,20 @@ def _get_artifact_repo(run):
     return get_artifact_repository(run.info.artifact_uri, store)
 
 
+@catch_mlflow_exception
+def _log_batch():
+    request_message = _get_request_message(ListExperiments())
```
Comment (mparkhe) [inline mlflow/server/handlers.py]: This should be -

```
    request_message = _get_request_message(LogBatch())
```

**Review by mparkhe — CHANGES_REQUESTED** [review]
copy paste error in `_log_batch` handler.

File: tests/store/test_rest_store.py [hunk 2]
```diff
@@ -105,6 +105,19 @@ def test_requestor(self, request):
             self._verify_requests(mock_http, creds,
                                   "runs/log-metric", "POST", body)
 
+        with mock.patch('mlflow.store.rest_store.http_request_safe') as mock_http:
```
Comment (mparkhe) [inline tests/store/test_rest_store.py]: let's add example(s) with real batches.
- a bunch of just metrics, tags and params are empty
- just tags,
- one tag, and a bunch of params
- params and tags with the same key
- metric and param batches swapped in API call, and other such error cases
- ... etc

ideally, we should have a generic test that is super comprehensive with all these options and applied to a any backend and ensure that the behavior is as intended

**Comment by smurching** [comment]
@mparkhe as per offline discussion, I ended up adding a couple of simple test cases to more thoroughly verify the end-to-end server functionality (i.e. actually integration-test against a real AbstractStore implementation instead of just verifying that the right AbstractStore methods are called).

Also created https://github.com/mlflow/mlflow/issues/984 to track adding a new test harness for verifying AbstractStore implementations (I think a good way to do this could be to just refactor the existing FileStore & SqlAlchemyStore unit tests into a single test module parametrized by store type)

File: tests/server/test_handlers.py [hunk 3]
```diff
@@ -78,6 +81,78 @@ def test_search_runs_default_view_type(mock_get_request_message, mock_store):
...
+    tag_entities_dict = [dict(t) for t in tag_entities]
+    approx_expected_tags = [MLFLOW_SOURCE_NAME, MLFLOW_SOURCE_TYPE]
+    for t in run.data.tags:
+        if t.key in approx_expected_tags:
+            pass
+        else:
+            assert dict(t) in tag_entities_dict
+
+
+def test_log_batch_handler_success(mock_get_request_message, tmpdir):
```
Comment (mparkhe) [inline tests/server/test_handlers.py]: this is really an awesome test -- at some point we can use the `@pytest.mark.parametrize` to send it different stores.

**Review by mparkhe — APPROVED** [review]
LGTM

**Pull request merged**